web/src/beta/lib/reearth-ui/components/Popup/hook.ts

```diff
@@ -0,0 +1,67 @@
+import {
+  useFloating,
+  autoUpdate,
+  offset,
+  flip,
+  shift,
+  useClick,
+  useDismiss,
+  useRole,
+  useInteractions,
+} from "@floating-ui/react";
+import { useMemo, useState } from "react";
+
+import { PopupOptionsProps } from ".";
+
+const usePopover = ({
+  placement = "bottom",
+  modal,
+  open: controlledOpen,
+  offset: offsetProps,
+  shift: shiftProps,
+  onOpenChange: setControlledOpen,
+}: PopupOptionsProps = {}) => {
+  const [uncontrolledOpen, setUncontrolledOpen] = useState(false);
+
+  const open = controlledOpen ?? uncontrolledOpen;
+  const setOpen = setControlledOpen ?? setUncontrolledOpen;
```

Comment:
I think we need to make these two as a pair, otherwise for example it could using controlledOpen and setUncontrolledOpen together.